utilities/network.py

```diff
@@ -1038,23 +1039,146 @@ def get_cluster_cni_type(admin_client):
+
+    if not _node_matches_policy_selector(node=node, policy=policy):
+        LOGGER.info(f"Skipping verification for node {node.name} - does not match policy {policy.name} nodeSelector")
+        return
+
+    policy_spec = policy.instance.spec
+    pf_names = policy_spec.nicSelector.get("pfNames", [])
+    root_devices = policy_spec.nicSelector.get("rootDevices", [])
+    expected_num_vfs = policy_spec.get("numVfs")
+    expected_mtu = policy_spec.get("mtu")
+
+    matching_interfaces = []
+    for iface in sriov_node_network_state.instance.status.interfaces:
+        if iface.name in pf_names or iface.pciAddress in root_devices:
+            matching_interfaces.append(iface)
+
+    if not matching_interfaces:
+        LOGGER.info(
+            f"Skipping verification for node {sriov_node_network_state.name} - "
+            f"no interfaces matching policy {policy.name} (pfNames={pf_names}, rootDevices={root_devices})"
+        )
+        return
+
+    for iface in matching_interfaces:
+        LOGGER.info(
+            f"Verifying interface {iface.name} on node {sriov_node_network_state.name}: "
+            f"numVfs={iface.numVfs} (expected: {expected_num_vfs})"
+        )
+
+        if iface.numVfs != expected_num_vfs:
+            LOGGER.error(
+                f"Interface {iface.name} on {sriov_node_network_state.name} has numVfs={iface.numVfs}, "
+                f"expected {expected_num_vfs}"
+            )
+            raise AssertionError(f"numVfs mismatch for {iface.name}: got {iface.numVfs}, expected {expected_num_vfs}")
+
+        if expected_mtu:
+            iface_mtu = getattr(iface, "mtu", None)
+            if iface_mtu and iface_mtu != expected_mtu:
+                LOGGER.error(
+                    f"Interface {iface.name} on {sriov_node_network_state.name} has mtu={iface_mtu}, "
+                    f"expected {expected_mtu}"
+                )
+                raise AssertionError(f"MTU mismatch for {iface.name}: got {iface_mtu}, expected {expected_mtu}")
+
+        LOGGER.info(f"Interface {iface.name} configuration verified successfully")
+
+
+def wait_for_ready_sriov_nodes(snns: list[SriovNetworkNodeState], policy: SriovNetworkNodePolicy = None):
+    """
+    Wait for SR-IOV nodes to be ready and verify interface configuration matches policy.
+
+    Args:
+        snns: List of SriovNetworkNodeState objects
+        policy: Optional SriovNetworkNodePolicy object to verify configuration against
+
+    Raises:
+        TimeoutExpiredError: If sync status doesn't reach SUCCEEDED state or configuration doesn't match
+        AssertionError: If interface configuration doesn't match policy
+    """
+    for sriov_node_network_state in snns:
+        LOGGER.info(f"Waiting for node {sriov_node_network_state.name} to reach SUCCEEDED status")
+        sampler = TimeoutSampler(
+            wait_timeout=1000,
+            sleep=5,
+            func=lambda snns=sriov_node_network_state: snns.instance.status.syncStatus,
+        )
```

Comment:
Moving the status check into a separate sampler doesn't solve the risk I raised in my [previous comment](https://github.com/RedHatQE/openshift-virtualization-tests/pull/2760#discussion_r2546315794).
If the SNNS status still reports the "old" `SUCCEEDED` status then the sampler will immediately return and falsely declare a success.
Verifying both the status and the actual configuration **together** in each iteration of the same sampler will achieve that. That's what I meant in that comment, so my apologies if I wasn't clear about it.
And it will also resolve @rnetser 's concern about the function being too long.